File: experiments/test_motors/test_motors.py
from device import Motor, RotaryEncoder, set_up, tear_down
from time import sleep

LEFT_MOTOR_DIRECTION_PIN = 6
LEFT_MOTOR_ENABLE_PIN = 13

LEFT_ENCODER_A_PIN = 9
LEFT_ENCODER_B_PIN = 11

ENCODER_RESOLUTION = 32
GEAR_RATIO = 344.2
WHEEL_TO_MOTOR_RATIO = ENCODER_RESOLUTION * GEAR_RATIO
MOTOR_TO_WHEEL_RATIO = WHEEL_TO_MOTOR_RATIO ** -1

SPEED = 1.0
DURATION = 5

if __name__ == "__main__":
    set_up()
    
    left_motor = Motor(LEFT_MOTOR_DIRECTION_PIN, LEFT_MOTOR_ENABLE_PIN)
    left_encoder = RotaryEncoder(LEFT_ENCODER_A_PIN, LEFT_ENCODER_B_PIN)
    
    left_motor.setSpeed(SPEED)

    sleep(DURATION)

    left_motor.setSpeed(0)
    
    left_step_count = left_encoder.getCount()
    left_turn_count = left_step_count * MOTOR_TO_WHEEL_RATIO
    
    print(f"step_count: {left_step_count}")
    print(f"turn_count: {left_turn_count}")

File: .old/experiments/test_motors/test_motors.cpp
#include <pigpio.h>
#include <unistd.h>

#include <iostream>

#include "Motor.hpp"
#include "RotaryEncoder.hpp"

int main(void)
{
    if (gpioInitialise() < 0) {
        return 1;
    }

    {
        Motor left_motor(6, 13);
        RotaryEncoder left_encoder(9, 11);

        left_motor.setSpeed(1.0);

        sleep(5);

        std::cout << left_encoder.getCount() << std::endl;

        const double MOTOR_TO_GEAR_RATIO = 1.0 / (32.0 * 344.2);
        std::cout << left_encoder.getCount() * MOTOR_TO_GEAR_RATIO << std::endl;
    }

    gpioTerminate();
}
